src/models/fields/base-field-configuration.ts

```diff
@@ -1,39 +1,358 @@
+      text: stringOrTranslationsSchema.required(),
+      value: Joi.number().integer().required().description('Will be passed through parser'),
+    })),
+    inputType: Joi.string().valid('radio').required(),
+  }).required(),
+  commonSchema.keys({
+    type: Joi.string().valid('boolean').required(),
+    defaultValue: Joi.boolean().default(false),
+    valueOptions: Joi.array().length(0).allow(null).default(null),
+    inputType: Joi.number().valid('switch', 'checkbox').default('checkbox'),
+  }).required(),
+  commonSchema.keys({
+    type: Joi.string().valid('boolean').required(),
+    defaultValue: Joi.boolean().default(0),
+    valueOptions: Joi.array().items(Joi.object().keys({
+      text: stringOrTranslationsSchema.required(),
+      value: Joi.boolean().required().description('Will be passed through parser'),
+    })),
+    inputType: Joi.string().valid('select').default('select'),
+    ...prefixesMixin,
+  }).required(),
+  commonSchema.keys({
+    type: Joi.string().valid('boolean').required(),
+    defaultValue: Joi.boolean().default(0),
+    valueOptions: Joi.array().items(Joi.object().keys({
+      text: stringOrTranslationsSchema.required(),
+      value: Joi.boolean().required().description('Will be passed through parser'),
+    })),
+    inputType: Joi.string().valid('radio').required(),
+  }).required(),
+  commonSchema.keys({
+    inputType: Joi.string().valid('file', 'files').required(),
+    valueOptions: Joi.array().length(0).allow(null),
+  }).required(),
+)
   .tag('baseFieldConfiguration')
   .description('Defines which data can be stored in form fields.');
 
-interface ValueOption {
-  text: string;
-  value: BaseField;
+const schema = getBaseFieldConfigurationSchema(commonBaseFieldConfigurationSchema);
+
+interface ValueOption<T extends BaseField> {
+  text: StringOrTranslations;
+  value: T;
 }
 
-interface BaseFieldConfiguration {
+interface SharedBaseFieldConfiguration {
+  /**
+   * key used to store the field value
+   * @pattern ^[a-z][a-zA-Z\d]*$
+   */
   key: string;
-  name: string;
-  inputType?: 'text' | 'textarea' | 'select' | 'radio' | 'switch' | 'checkbox' | 'file' | 'files';
-  valueOptions?: ValueOption[] | null;
-  prefix?: string;
-  suffix?: string;
-  hint?: string;
+  /**
+   * name of the field to be shown in UI
+   */
+  name: StringOrTranslations;
+  /**
+   * show this field in UI only if value of an earlier form field, identified by provided `key`, is
+   * equal to provided `value`
+   */
+  showIf?: { key: string, value: string | boolean | number };
+  hint?: StringOrTranslations;
 }
 
+interface PrefixMixin {
+  prefix?: StringOrTranslations;
+  suffix?: StringOrTranslations;
+}
+
+type integer = number;
+
+type BaseFieldConfiguration = ({
+  type: 'string';
+  /**
+   * @default ""
+   */
+  defaultValue: StringOrTranslations;
+  /**
+   * @default null
+   */
+  valueOptions: null | [];
+  /**
+   * @default "text"
+   */
+  inputType: 'text' | 'textarea' | 'translations';
```

Comment:
What's the  `'translations'` type?

Reply:
I've added such input type to display `translation-text-field` in `editable-field `, and thought it would be nice to add it here as well.

Reply:
I don't think that works: a filled out translation-text-field would produce data that does not conform to `BaseField` but is an object or array itself. 

(it's not in the Joi schema by the way).

Reply:
Fixed schema, added `Translations` to `BaseField` (including Joi schema)

Reply:
Yeah, but that means that we'll be storing complexer objects in the `fields` columns in the database, while right now they are always objects that are one level deep. I'm only for increasing complexity if the benefit is clear. I don't really see a good use case for this, what do you have in mind?

Reply:
I was thinking so: string value can be constant (identifier, serial number, hash, etc) or it can be translateable (name, description). 

Now I've reviewed few production environments and see that all actual fields string values do not need to be translated.

So I agree that users do not really need it and will remove it now from `BaseFieldConfiguration` and remove `Translations` from `BaseField` union.